server/src/main/java/org/opensearch/index/shard/ReplicationGroup.java

```diff
@@ -49,22 +50,32 @@ public class ReplicationGroup {
     private final IndexShardRoutingTable routingTable;
     private final Set<String> inSyncAllocationIds;
     private final Set<String> trackedAllocationIds;
+    private final Set<String> localTranslogAllocationIds;
     private final long version;
+    private final boolean remoteTranslogEnabled;
 
     private final Set<String> unavailableInSyncShards; // derived from the other fields
-    private final List<ShardRouting> replicationTargets; // derived from the other fields
+    private final List<ReplicationAwareShardRouting> replicationTargets; // derived from the other fields
     private final List<ShardRouting> skippedShards; // derived from the other fields
 
     public ReplicationGroup(
         IndexShardRoutingTable routingTable,
         Set<String> inSyncAllocationIds,
         Set<String> trackedAllocationIds,
-        long version
+        Set<String> localTranslogAllocationIds,
+        long version,
+        boolean remoteTranslogEnabled
     ) {
+        if (!remoteTranslogEnabled) {
+            assert localTranslogAllocationIds.containsAll(trackedAllocationIds)
```

Comment:
Why not `equals`?

Reply:
when a shard is initialising, it is not tracked. In that case, the `localTranslogAllocationIds` would have reference to shard, but `trackedAllocationIds` wont.